### DISCUSSION TITLE
Use meta route outside pages component

### DISCUSSION THREAD

**Discussion by fmgono** [nuxt/nuxt#8914] [General]
I have a problem in nuxt js, because I want to implement the router meta for consumption by other components outside the pages folder
Here is an example :

```js
// in pages/dashboard.vue
export default {
 meta: {
  pageTitle: 'Dashboard'
 }
}
```

and i want to access this title in Navbar.vue in `components/layout/Navbar.vue`
(show it as title in Navbar)

```vue
// in components/layout/Navbar.vue template
<template>
 <h3 v-if="$route.meta.pageTitle">{{ $route.meta.pageTitle }}</h3>
</template>
```

how can i achive this in nuxt ? because this meta file could be help a lot for roles, route validations, etc

**Comment by udithishara** [comment]
@fmgono I've used a similar thing before, but using `head()`, Downside is you have to pass the data into the component specifically from the pages

```
// in pages/index.vue script

import Navbar from '@/components/Navbar'

export default {
  components: {
    Navbar,
  },
  head() {
    return {
      title: 'Dashboard',
    }
  },
}
```

```
// in pages/index.vue template

<Navbar :meta-title="$metaInfo.title" />
```

```
// in components/Navbar.vue

<template>
  <h3>{{ metaTitle }}</h3>
</template>

<script>
export default {
  props: {
    metaTitle: {
      type: String,
      required: true,
    },
  },
}
</script>
```

**Comment by SKhler** [comment]
I come to the Bank (Sparkasse)


Gesendet von Yahoo Mail für iPhone


Am Samstag, März 6, 2021, 6:48 AM schrieb Udith Ishara Madusanka <notifications@github.com>:




@fmgono I've used a similar thing before, but using head(), Downside is you have to pass the data into the component specifically from the pages
// in pages/index.vue script

import Navbar from '@/components/Navbar'

export default {
  components: {
    Navbar,
  },
  head() {
    return {
      title: 'Dashboard',
    }
  },
}
// in pages/index.vue template

<Navbar :meta-title="$metaInfo.title" />
// in components/Navbar.vue

<template>
  <h3>{{ metaTitle }}</h3>
</template>

<script>
export default {
  props: {
    metaTitle: {
      type: String,
      required: true,
    },
  },
}
</script>

**Comment by vhoyer** [comment]
But route's `meta` is accessible from every component...

Maybe because it's being defined from the component it may be working correctly? I donno, what I do know is that I use the `meta` on the router to dynamically construct the website's breadcrumb, and I do access it through `this.$route.meta` inside the `breadcrumb` component. The only difference I could figure is that I'm defining the routes manually using the [`extendRoutes`](https://nuxtjs.org/docs/2.x/configuration-glossary/configuration-router/#extendroutes), have you tried doing it?

If you don't want to manually define all the routes, maybe you could try doing something dynamic like:

```js
export default {
  router: {
    extendRoutes(routeList, resolve) {
      routeList.forEach(route => {
        route.meta = require(resolve(route.component)).meta;
      })
    },
  }
}
```

As Far as I know, this will not be a problem regarding loading all components files, because this `extendRoutes` method runs on the build, not on the front end (But I'd not believe a random on the internet, you should test it for yourself :D).

**Reply by changemyminds** [reply]
@vhoyer  When I create new project and add this code in the nuxt.config.js, I get the error.

```javascript
export default {   
  ssr: false, 
  router: {
    extendRoutes(routes, resolve) {
      routes.forEach(route => {
        route.meta = require(resolve(route.component)).meta;
      })
    },
  }
}
```

The error code is below.  Am I missing something? Seems the problem is `require` method is not working correct.
```
 ERROR  Unexpected token '<'                                                                                                                                                                15:53:46  

  (function (exports, require, module, __filename, __dirname) { <template>
  ^

  SyntaxError: Unexpected token '<'
  at new Script (vm.js:102:7)
  at p._moduleCompile (node_modules\jiti\dist\v8cache.js:2:3154)
  at Module.o._compile (node_modules\jiti\dist\v8cache.js:2:2705)
  at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
  at Module.load (internal/modules/cjs/loader.js:937:32)
  at Function.Module._load (internal/modules/cjs/loader.js:778:12)
  at Module.require (internal/modules/cjs/loader.js:961:19)
  at require (internal/modules/cjs/helpers.js:92:18)
  at v (node_modules\jiti\dist\jiti.js:1:53966)
  at nuxt.config.js:50:22
  at Array.forEach (<anonymous>)
  at Object.extendRoutes (nuxt.config.js:49:14)
  at Builder.resolveRoutes (node_modules\@nuxt\builder\dist\builder.js:546:56)
  at async Promise.all (index 1)
  at async Builder.generateRoutesAndFiles (node_modules\@nuxt\builder\dist\builder.js:375:5)
  at async Builder.build (node_modules\@nuxt\builder\dist\builder.js:319:5)
```
